examples/device-dashboard/net.c

```diff
@@ -132,7 +132,11 @@ static void mqtt_fn(struct mg_connection *c, int ev, void *ev_data, void *fnd) {
   } else if (ev == MG_EV_MQTT_OPEN) {
     s_connected = true;
     c->is_hexdumping = 1;
-    struct mg_mqtt_opts sub_opts = {.topic = mg_str(s_config.sub), .qos = 2};
+    struct mg_mqtt_opts sub_opts;
+    memset(&sub_opts, 0, sizeof(sub_opts));
+    sub_opts.topic = mg_str(s_config.sub);
+    sub_opts.qos = 2;
+    
```

Comment:
Hi, sorry for breaking into. Is there any good reason for changing this ? We have tutorials referring this piece of code that will have to be updated, it is passing all our current tests so I don't think there are compiler issues, or are they ?

Reply:
Designated initializers could be an issue with the older compilers (like VC98, which we traditionally support), and some C++ compilers.